samples/src/main/java/org/opensearch/client/samples/Bulk.java

```diff
@@ -70,21 +73,46 @@ public static void main(String[] args) {
             BulkResponse bulkResponse = client.bulk(bulkReq.build());
             LOGGER.info("Bulk response items: {}", bulkResponse.items().size());
 
-            Query query = Query.of(qb -> qb.match(mb -> mb.field("title").query(fv -> fv.stringValue("Document"))));
-            final SearchRequest.Builder searchReq = new SearchRequest.Builder().allowPartialSearchResults(false)
-                .index(List.of(indexName))
-                .size(10)
-                .source(sc -> sc.fetch(false))
-                .ignoreThrottled(false)
-                .query(query);
-            SearchResponse<IndexData> searchResponse = client.search(searchReq.build(), IndexData.class);
+            // wait for the changes to reflect
+            Thread.sleep(3000);
+            LOGGER.info("Search & bulk update documents");
+
+            SearchResponse<IndexData> searchResponse = search(client, indexName, "title", "Document");
             LOGGER.info("Found {} documents", searchResponse.hits().hits().size());
 
+            for (var hit : searchResponse.hits().hits()) {
```

Comment:
You could just use the update for a particular document directly since every bulk item has an `id`:

```suggestion
            BulkRequest request = new BulkRequest.Builder().operations(
                    o -> o.update(u -> u.index(indexName).id("id1").document(new IndexData("Document 1", "Updated text of document 1")))
                ).build();
                bulkResponse = client.bulk(request);

```

Reply:
This is exactly the show case: update existing document, we have an `id` and new content